model_compression_toolkit/core/keras/quantizer/lut_fake_quant.py

```diff
@@ -0,0 +1,118 @@
+
+    Returns:
+        A fake LUT quantization node.
+    """
+
+    lut_fake_quant = LUTFakeQuant(quantization_params=quantization_params)
+    return lambda x: lut_fake_quant(x)
+
+
+class LUTFakeQuant(Layer):
+    """
+    A custom Keras layer for quantizing activation tensor with non-uniform quantization (using lookup table clustering).
+    """
+
+    def __init__(self, quantization_params: Dict[str, np.ndarray], **kwargs):
+        super(LUTFakeQuant, self).__init__(**kwargs)
+        self.quantization_params = quantization_params
+        self.activation_is_signed = self.quantization_params.get(SIGNED)
+        self.cluster_centers = self.quantization_params.get(CLUSTER_CENTERS)
+        self.threshold = self.quantization_params.get(THRESHOLD)
+
+    def build(self, input_shape: Tuple[int]):
+        """
+        Builds the layer.
+
+        Args:
+            input_shape: The layer's input shape.
+
+        """
+        super(LUTFakeQuant, self).build(input_shape)
+
+    def call(self, input_data: TFReference, **kwargs) -> TFReference:
+        """
+
+        Args:
+            input_data: A Keras input tensor.
+            **kwargs: Optional arguments' dictionary.
+
+        Returns: KerasTensor after applying a non-uniform fake quantization.
+
+        """
+        if self.activation_is_signed is None or self.cluster_centers is None or self.threshold is None:
+            return None
+
+        _quant_output = self.lut_kmeans_quantizer(input_data)
+        return _quant_output
+
+    def lut_kmeans_quantizer(self, tensor_data: TFReference) -> TFReference:
+        """
+        Quantize a tensor using a non-uniform quantization based on the pre-defined kmeans clusters.
+        1. Scales tensor_data with the threshold into 8-bit quantization range.
+        2. Assigns cluster centers to each value.
+        3. Scales back by multiplying the result by threshold and dividing with the quantization range max value.
+        The result is the quantized tensor.
+
+        Args:
+            tensor_data: Input activation tensor.
+
+        Returns: Quantized tensor.
+        """
+
+        tensor = self.int_quantization_with_threshold(tensor_data, MULTIPLIER_N_BITS)
+        tensor = tf.expand_dims(tensor, -1)
+
+        expanded_cluster_centers = self.cluster_centers.reshape([*[1 for _ in range(len(tensor.shape)-1)], -1])
+        cluster_assignments = tf.argmin(tf.abs(tensor - expanded_cluster_centers), axis=-1)
+        centers = tf.gather(self.cluster_centers.flatten(), cluster_assignments)
+
+        quant_tensor = (centers / (2 ** (MULTIPLIER_N_BITS - int(self.activation_is_signed)))) * self.threshold
+
+        return quant_tensor
+
+    def int_quantization_with_threshold(self,
+                                        data: TFReference,
+                                        n_bits: int,
+                                        eps: float = EPS) -> TFReference:
+        """
```

Comment:
Why don't we use int_quantization_with_threshold that is in quantizers_helpers?

Reply:
Because it is works on np arrays and we need an implementation for tf.Tensors

Reply:
Ok. So I just think maybe it's better to create a similar file (like quantizers utils) for TF quantizers, as this it shouldn't be coupled to LUT specifically. What do you think?